src/copy.c

```diff
@@ -542,7 +544,8 @@ timescaledb_DoCopy(const CopyStmt *stmt, const char *queryString, uint64 *proces
 	Assert(rel);
 
 	/* check read-only transaction and parallel mode */
-	if (XactReadOnly && !rel->rd_islocaltemp)
+	xactReadOnly = GetConfigOptionByName("transaction_read_only", NULL, false);
+	if (strncmp(xactReadOnly, "on", sizeof("on")) == 0 && !rel->rd_islocaltemp)
```

Comment:
should you check for xactReadOnly != NULL?

Reply:
Don't think it's necessary since `missing_ok` is set to false, so if it's not there this part errors before that check.